cli/cmd/metrics_diagnostics_util.go

```diff
@@ -125,12 +128,13 @@ func getMetrics(
 		}(pod)
 	}
 
+wait:
 	for {
 		select {
 		case result := <-resultChan:
 			results = append(results, result)
 		case <-time.After(waitingTime):
-			break // timed out
+			break wait // timed out
```

Comment:
Hm not sure how we haven't caught this before. Thanks!

Reply:
I expected the linter to pick it up, but upon inspection the `staticcheck` linter is explicitly disabled for this repo.

https://github.com/linkerd/linkerd2/blob/c031939ea4af873ec85a8c00a651081f7d915de6/.golangci.yml#L43-L46